plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.trino.hdfs.ConfigurationUtils;
+import io.trino.plugin.base.CatalogName;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.iceberg.IcebergConfig;
+import io.trino.plugin.iceberg.catalog.TrinoCatalog;
+import io.trino.plugin.iceberg.catalog.TrinoCatalogFactory;
+import io.trino.spi.security.ConnectorIdentity;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.rest.RESTSessionCatalog;
+import org.apache.iceberg.rest.auth.OAuth2Properties;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class TrinoIcebergRestCatalogFactory
+        implements TrinoCatalogFactory
+{
+    private static final Logger log = Logger.get(TrinoIcebergRestCatalogFactory.class);
+
+    private final CatalogName catalogName;
+    private final String trinoVersion;
+    private final URI serverUri;
+    private final Optional<String> credential;
+    private final Optional<String> token;
+    private final IcebergRestCatalogConfig.Security security;
+    private final boolean isUniqueTableLocation;
+
+    private volatile RESTSessionCatalog icebergCatalog;
+
+    @Inject
+    public TrinoIcebergRestCatalogFactory(
+            CatalogName catalogName,
+            IcebergRestCatalogConfig restConfig,
+            IcebergConfig icebergConfig,
+            NodeVersion nodeVersion)
+    {
+        this.catalogName = requireNonNull(catalogName, "catalogName is null");
+        this.trinoVersion = requireNonNull(nodeVersion, "nodeVersion is null").toString();
+        requireNonNull(restConfig, "restConfig is null");
+        this.serverUri = restConfig.getBaseUri();
+        this.credential = restConfig.getCredential();
+        this.token = restConfig.getToken();
+        this.security = restConfig.getSecurity();
+        requireNonNull(icebergConfig, "icebergConfig is null");
+        this.isUniqueTableLocation = icebergConfig.isUniqueTableLocation();
+    }
+
+    @Override
+    public TrinoCatalog create(ConnectorIdentity identity)
+    {
+        if (icebergCatalog == null) {
+            synchronized (this) {
+                if (icebergCatalog == null) {
+                    ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
+                    properties.put(CatalogProperties.URI, serverUri.toString());
+                    properties.put(CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.io.ResolvingFileIO");
```

Comment:
`ResolvingFileIO.class.getName()`

btw why this one? please document.

Reply:
I was able to remove that since it is now the default in the REST Catalog.  This is just a FileIO that will resolve the IO backend based on the scheme of the file paths.